packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -513,7 +513,7 @@ class RenderParagraph extends RenderBox
   void _setParentData() {
     RenderBox child = firstChild;
     int childIndex = 0;
-    while (child != null) {
+    while (child != null && childIndex < _textPainter.inlinePlaceholderBoxes.length) {
```

Comment:
do we want to set the offsets to null for the children that aren't being laid out this time? otherwise we'll have out of date offsets

Reply:
is there a situation that we use offsets for children that are not painted? It sounds to me that we should really remove this children if it is not painted.